lib/ladybug/energy_model/face.rb

```diff
@@ -0,0 +1,149 @@
+
+require 'ladybug/energy_model/model_object'
+require 'ladybug/energy_model/aperture'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class Face < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash = {})
+        super(hash)
+
+        raise "Incorrect model type '#{@type}'" unless @type == 'Face'
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        model_surf = openstudio_model.getSurfaceByName(@hash[:name])
+        return model_surf.get unless model_surf.empty?
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)       
+        openstudio_vertices = OpenStudio::Point3dVector.new
+        @hash[:geometry][:boundary].each do |vertex|
+          openstudio_vertices << OpenStudio::Point3d.new(vertex[0], vertex[1], vertex[2])
+        end
+
+        openstudio_construction = nil
+        if @hash[:properties][:energy][:construction]
+          construction_name = @hash[:properties][:energy][:construction]
+          construction = openstudio_model.getConstructionByName(construction_name)
+          unless construction.empty?
+            openstudio_construction = construction.get
+          end
+        end
+
+        openstudio_surface = OpenStudio::Model::Surface.new(openstudio_vertices, openstudio_model)
+        openstudio_surface.setName(@hash[:name])
+        openstudio_surface.setSurfaceType(@hash[:face_type])
+        openstudio_surface.setConstruction(openstudio_construction) if openstudio_construction
+        openstudio_surface.setOutsideBoundaryCondition(@hash[:boundary_condition][:type])
+        boundary_condition = (@hash[:boundary_condition][:type])
+       
+
+        case boundary_condition
+        when 'Outdoors'
+          if @hash[:boundary_condition][:sun_exposure] == true
+            openstudio_surface.setSunExposure('SunExposed')
+          else 
+            openstudio_surface.setSunExposure('NoSun')
+          end
+          if @hash[:boundary_condition][:wind_exposure] == true
+            openstudio_surface.setWindExposure('WindExposed')
+          else
+            openstudio_surface.setWindExposure('NoWind')
+          end
+          if @hash[:boundary_condition][:view_factor] == 'autocalculate'
+            openstudio_surface.autocalculateViewFactortoGround
+          else
+            openstudio_surface.setViewFactortoGround(@hash[:boundary_condition][:view_factor])
+          end
+        when 'Surface'
+          surface = nil 
+          if @hash[:boundary_condition][:boundary_condition_objects][0]
+            surface_object = openstudio_model.getSurfaceByName(@hash[:boundary_condition][:boundary_condition_objects][0])
+            unless surface_object.empty?
+              surface = surface_object.get
+              openstudio_surface.setAdjacentSurface(surface)
```

Comment:
I think that you will also need the following code here:
```
when 'Adiabatic
... hard assign the Construction to the Face using the Interior construction from the ConstructionSet.
```
I guess you can also do this after you have created all of the faces in the model.